src/table/TableBodyWrapper.jsx

```diff
@@ -26,21 +26,20 @@ const TableBodyWrapper = ({ rootElement, tableData, constraints, selectionsAPI,
   const hoverEffect = layout.components?.[0]?.content?.hoverEffect;
   const styling = useMemo(() => getBodyStyle(layout, theme), [layout, theme.name()]);
   const classes = useStyles(styling);
-  const getColumnRenderers = (selectionsEnabled) =>
-    tableData.columns.map((c) => getCellRenderer(!!c.stylingInfo.length, selectionsEnabled));
-  const [columnRenderers, setColumnRenderers] = useState(getColumnRenderers(false));
+  const selectionsEnabled = !!selectionsAPI && !constraints.active;
+  const getColumnRenderers = tableData.columns.map((c) => getCellRenderer(!!c.stylingInfo.length, selectionsEnabled));
+  const [columnRenderers, setColumnRenderers] = useState(() => getColumnRenderers);
   const [selState, selDispatch] = useReducer(reducer, {
     api: selectionsAPI,
     rows: [],
     colIdx: -1,
-    isEnabled: !!selectionsAPI && !constraints.active,
+    isEnabled: selectionsEnabled,
   });
 
   useEffect(() => {
-    const selectionsEnabled = !!selectionsAPI && !constraints.active;
     selDispatch({ type: 'set-enabled', payload: { isEnabled: selectionsEnabled } });
-    setColumnRenderers(getColumnRenderers(selectionsEnabled));
-  }, [constraints, columns.length]);
+    setColumnRenderers(getColumnRenderers);
+  }, [selectionsEnabled, columns.length]);
```

Comment:
Hmm, is selectionsEnabled kept between renders? Otherwise it will change every render and this hook runs every time?

Reply:
the selectionsEnabled is same between renders. I think that selectionsEnabled was changed between analysis mode and edit mode.